pipelines/batch/src/main/java/org/openmrs/analytics/ReadJsonFilesFn.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package org.openmrs.analytics;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Set;
+
+import ca.uhn.fhir.parser.DataFormatException;
+import com.google.common.collect.Sets;
+import org.apache.beam.sdk.io.FileIO;
+import org.apache.beam.sdk.values.KV;
+import org.hl7.fhir.instance.model.api.IBaseReference;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Bundle.BundleEntryComponent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ReadJsonFilesFn extends FetchSearchPageFn<FileIO.ReadableFile> {
+	
+	private static final Logger log = LoggerFactory.getLogger(ReadJsonFilesFn.class);
+	
+	private final Set<String> resourceTypes;
+	
+	ReadJsonFilesFn(FhirEtlOptions options) {
+		super(options, "ReadJsonFiles");
+		resourceTypes = Sets.newHashSet(options.getResourceList().split(","));
+	}
+	
+	@ProcessElement
+	public void processElement(@Element FileIO.ReadableFile file, OutputReceiver<KV<String, Integer>> out)
+	        throws IOException, SQLException {
+		log.info("Reading file with metadata " + file.getMetadata());
+		String fileContent = file.readFullyAsUTF8String();
+		try {
+			IBaseResource resource = parser.parseResource(fileContent);
+			if (!"Bundle".equals(resource.fhirType())) {
+				log.error("The content of file " + file.getMetadata().toString() + " is not a Bundle; type: "
```

Comment:
If it is better practice, switch to using `String.format` as opposed to string concatenation